### ISSUE TITLE
Rolemanager doesn't add buttons and reactions + embeds sometimes not generated

### ISSUE THREAD

**Issue by RFlintstone** [Jerrie-Aries/modmail-plugins#14]
I found rolemanager and the description was exactly what I was looking for however I find
when I create embeds with [p]reactrole for manual roles it never seems to add buttons or reactions.

I tried all commands and the following commands didn't give back an embed and don't seem to work ([p] being the command prefix):

### **[p]autorole**
[p]autorole add (only if the role exists)
[p]autorole enable
[p]autorole list

### **[p]reactrole**
- [p]enable
- [p]list
- [p]refresh
- [p]repair

Is this plugin still under development, or is there something wrong with a dependency?
Either way, cool plugin! 👌

**Comment by Jerrie-Aries** [comment]
The `master` branch should work.

Is there any error, can you post it if any?
Also it is required to install the [utils](https://github.com/Jerrie-Aries/modmail-plugins/tree/master/utils) plugin first.

**Comment by RFlintstone** [comment]
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/discord/client.py", line 409, in _run_event
    await coro(*args, **kwargs)
  File "/modmailbot/plugins/Jerrie-Aries/modmail-plugins/rolemanager-master/rolemanager.py", line 1377, in on_raw_message_delete
    if not self.config.reactroles.find_entry(payload.message_id):
AttributeError: '_MissingSentinel' object has no attribute 'reactroles'
10/25/22 12:32:37 __main__[1500] - ERROR: Ignoring exception in on_raw_message_delete.
10/25/22 12:32:37 __main__[1501] - ERROR: Unexpected exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/discord/client.py", line 409, in _run_event
    await coro(*args, **kwargs)
  File "/modmailbot/plugins/Jerrie-Aries/modmail-plugins/rolemanager-master/rolemanager.py", line 1377, in on_raw_message_delete
    if not self.config.reactroles.find_entry(payload.message_id):
AttributeError: '_MissingSentinel' object has no attribute 'reactroles'
10/25/22 12:32:45 __main__[1548] - ERROR: Unexpected exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/discord/ext/commands/core.py", line 190, in wrapped
    ret = await coro(*args, **kwargs)
  File "/modmailbot/plugins/Jerrie-Aries/modmail-plugins/moderation-master/moderation.py", line 328, in mute
    await member.timeout(duration.dt, reason=get_audit_reason(ctx.author, reason))
  File "/usr/local/lib/python3.9/site-packages/discord/member.py", line 975, in timeout
    await self.edit(timed_out_until=timed_out_until, reason=reason)
  File "/usr/local/lib/python3.9/site-packages/discord/member.py", line 874, in edit
    data = await http.edit_member(guild_id, self.id, reason=reason, **payload)
  File "/usr/local/lib/python3.9/site-packages/discord/http.py", line 738, in request
    raise Forbidden(response, data)
discord.errors.Forbidden: 403 Forbidden (error code: 50013): Missing Permissions
10/25/22 12:32:50 __main__[1500] - ERROR: Ignoring exception in on_raw_message_delete.
10/25/22 12:32:50 __main__[1501] - ERROR: Unexpected exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/discord/client.py", line 409, in _run_event
    await coro(*args, **kwargs)
  File "/modmailbot/plugins/Jerrie-Aries/modmail-plugins/rolemanager-master/rolemanager.py", line 1377, in on_raw_message_delete
    if not self.config.reactroles.find_entry(payload.message_id):
AttributeError: '_MissingSentinel' object has no attribute 'reactroles'
10/25/22 12:33:35 __main__[1500] - ERROR: Ignoring exception in on_raw_message_delete.
10/25/22 12:33:35 __main__[1501] - ERROR: Unexpected exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/discord/client.py", line 409, in _run_event
    await coro(*args, **kwargs)
  File "/modmailbot/plugins/Jerrie-Aries/modmail-plugins/rolemanager-master/rolemanager.py", line 1377, in on_raw_message_delete
    if not self.config.reactroles.find_entry(payload.message_id):
AttributeError: '_MissingSentinel' object has no attribute 'reactroles'
10/25/22 12:33:35 __main__[1500] - ERROR: Ignoring exception in on_raw_bulk_message_delete.
10/25/22 12:33:35 __main__[1501] - ERROR: Unexpected exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/discord/client.py", line 409, in _run_event
    await coro(*args, **kwargs)
  File "/modmailbot/plugins/Jerrie-Aries/modmail-plugins/rolemanager-master/rolemanager.py", line 1389, in on_raw_bulk_message_delete
    reactrole = self.config.reactroles.find_entry(message_id)
AttributeError: '_MissingSentinel' object has no attribute 'reactroles'
10/25/22 12:33:49 __main__[1500] - ERROR: Ignoring exception in on_raw_message_delete.
10/25/22 12:33:49 __main__[1501] - ERROR: Unexpected exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/discord/client.py", line 409, in _run_event
    await coro(*args, **kwargs)
  File "/modmailbot/plugins/Jerrie-Aries/modmail-plugins/rolemanager-master/rolemanager.py", line 1377, in on_raw_message_delete
    if not self.config.reactroles.find_entry(payload.message_id):
AttributeError: '_MissingSentinel' object has no attribute 'reactroles'
10/25/22 12:33:49 __main__[1500] - ERROR: Ignoring exception in on_raw_message_delete.
10/25/22 12:33:49 __main__[1501] - ERROR: Unexpected exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/discord/client.py", line 409, in _run_event
    await coro(*args, **kwargs)
  File "/modmailbot/plugins/Jerrie-Aries/modmail-plugins/rolemanager-master/rolemanager.py", line 1377, in on_raw_message_delete
    if not self.config.reactroles.find_entry(payload.message_id):
AttributeError: '_MissingSentinel' object has no attribute 'reactroles'
10/25/22 12:33:49 __main__[1500] - ERROR: Ignoring exception in on_raw_bulk_message_delete.
10/25/22 12:33:49 __main__[1501] - ERROR: Unexpected exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/discord/client.py", line 409, in _run_event
    await coro(*args, **kwargs)
  File "/modmailbot/plugins/Jerrie-Aries/modmail-plugins/rolemanager-master/rolemanager.py", line 1389, in on_raw_bulk_message_delete
    reactrole = self.config.reactroles.find_entry(message_id)
AttributeError: '_MissingSentinel' object has no attribute 'reactroles'
10/25/22 12:33:58 __main__[1500] - ERROR: Ignoring exception in on_raw_message_delete.
10/25/22 12:33:58 __main__[1501] - ERROR: Unexpected exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/discord/client.py", line 409, in _run_event
    await coro(*args, **kwargs)
  File "/modmailbot/plugins/Jerrie-Aries/modmail-plugins/rolemanager-master/rolemanager.py", line 1377, in on_raw_message_delete
    if not self.config.reactroles.find_entry(payload.message_id):
AttributeError: '_MissingSentinel' object has no attribute 'reactroles'
10/25/22 12:33:58 __main__[1500] - ERROR: Ignoring exception in on_raw_bulk_message_delete.
10/25/22 12:33:58 __main__[1501] - ERROR: Unexpected exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/discord/client.py", line 409, in _run_event
    await coro(*args, **kwargs)
  File "/modmailbot/plugins/Jerrie-Aries/modmail-plugins/rolemanager-master/rolemanager.py", line 1389, in on_raw_bulk_message_delete
    reactrole = self.config.reactroles.find_entry(message_id)
AttributeError: '_MissingSentinel' object has no attribute 'reactroles'
10/25/22 12:34:11 __main__[1500] - ERROR: Ignoring exception in on_raw_message_delete.
10/25/22 12:34:11 __main__[1501] - ERROR: Unexpected exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/discord/client.py", line 409, in _run_event
    await coro(*args, **kwargs)
  File "/modmailbot/plugins/Jerrie-Aries/modmail-plugins/rolemanager-master/rolemanager.py", line 1377, in on_raw_message_delete
    if not self.config.reactroles.find_entry(payload.message_id):
AttributeError: '_MissingSentinel' object has no attribute 'reactroles'
10/25/22 12:41:18 __main__[1518] - WARNING: CommandNotFound: Command "warns" is not found
10/25/22 12:42:19 __main__[1518] - WARNING: CommandNotFound: Command "warnings" is not found
10/25/22 12:49:15 __main__[1500] - ERROR: Ignoring exception in on_raw_message_delete.
10/25/22 12:49:15 __main__[1501] - ERROR: Unexpected exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/discord/client.py", line 409, in _run_event
    await coro(*args, **kwargs)
  File "/modmailbot/plugins/Jerrie-Aries/modmail-plugins/rolemanager-master/rolemanager.py", line 1377, in on_raw_message_delete
    if not self.config.reactroles.find_entry(payload.message_id):
AttributeError: '_MissingSentinel' object has no attribute 'reactroles'
10/25/22 12:51:44 __main__[1500] - ERROR: Ignoring exception in on_raw_reaction_add.
10/25/22 12:51:44 __main__[1501] - ERROR: Unexpected exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/discord/client.py", line 409, in _run_event
    await coro(*args, **kwargs)
  File "/modmailbot/plugins/Jerrie-Aries/modmail-plugins/rolemanager-master/rolemanager.py", line 1370, in on_raw_reaction_add_or_remove
    await self.config.reactroles.handle_reaction(payload)
AttributeError: '_MissingSentinel' object has no attribute 'reactroles'
```

**Comment by Jerrie-Aries** [comment]
Ok. The `rolemanager` plugin needs to be reloaded (uninstall and reinstall) after the `utils` plugin error. The `utils` must be the one that is loaded before `rolemanager`. Also maybe restarting the bot would do the trick.

**Comment by RFlintstone** [comment]
I restarted the bot but the problem is still there

**Comment by RFlintstone** [comment]
Reinstalled the plugin as well, issue persists

**Comment by Jerrie-Aries** [comment]
Resolved in commit [6ef58e4d](https://github.com/Jerrie-Aries/modmail-plugins/commit/6ef58e4dd7d72c3670401a2d9faeea9f0ab97902).

The issue was fetching config from database (`Config.fetch()`) returning empty dictionary and the bot was trying to access the key with `__get__` (i.e. accessing using square brackets), resulting `KeyError` and the plugin not properly loaded.
